code/modules/power/solar.dm

```diff
@@ -285,13 +285,14 @@
 	autostart = 1 // Automatically start
 
 /obj/machinery/power/solar_control/Initialize()
-	..()
-	if(!powernet)
-		return
+	SSsun.solars |= src
+	. = ..()
+
+/obj/machinery/power/solar_control/proc/setup()
```

Comment:
How would this work for maploaded solars? I don't see a mechanism for calling setup outside of the sun subsystem startup

Reply:
My bad